src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```
@@ -227,7 +227,8 @@ services:
 
   PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType:
     arguments:
-      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\DefaultLanguageToFilledArrayDataTransformer'
+      $defaultLanguageToFilledArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\DefaultLanguageToFilledArrayDataTransformer'
+      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\StringArrayToIntegerArrayDataTransformer'
```

Wow great catch @Codencode 
I searched for the PR that modified this and found it here https://github.com/PrestaShop/PrestaShop/pull/31391/files#diff-ce2d57a2ac02f46a9cd2eea65b89142c51e7df0a68afc4f9d1e9e01a53999f75L351

So actually your PR should be even simpler, it's just that the wrong transformer was assigned during the refacto, the ShopChoiceTreeType has nothing to do with multi lang values so `DefaultLanguageToFilledArrayDataTransformer` shouldn't be passed at all

```suggestion
      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\StringArrayToIntegerArrayDataTransformer'
```

---

Hello @jolelievre,
thank you for the suggestion, but if I apply this change, I get the following error:
```php
Cannot autowire service "PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType":
argument "$defaultLanguageToFilledArrayDataTransformer" of method "__construct()"
 references interface "Symfony\Component\Form\DataTransformerInterface" but no such service exists.
  You should maybe alias this interface to one of these existing services:
 "PrestaShopBundle\Form\Admin\Sell\Product\DataTransformer\RedirectionTargetTransformer",
 "PrestaShopBundle\Form\Admin\Sell\Product\DataTransformer\SpecificPriceFixedPriceTransformer", 
 "PrestaShopBundle\Form\Admin\Type\ApeType", "PrestaShopBundle\Form\Admin\Type\IpAddressType", 
 "PrestaShopBundle\Form\Admin\Type\TextPreviewType", 
 "PrestaShopBundle\Form\DataTransformer\ArabicToLatinDigitDataTransformer", 
 "PrestaShopBundle\Form\DataTransformer\DefaultEmptyDataTransformer", 
 "PrestaShopBundle\Form\DataTransformer\DefaultLanguageToFilledArrayDataTransformer", 
 "PrestaShopBundle\Form\DataTransformer\IDNConverterDataTransformer", 
 "PrestaShopBundle\Form\DataTransformer\StringArrayToIntegerArrayDataTransformer".
```
This happens because the class `PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType` expects `$defaultLanguageToFilledArrayDataTransformer`.

Am I missing something?

---

@jolelievre,
sorry, the issue was due to the fact that I "obviously" need to remove the instructions related to the `defaultLanguageToFilledArrayDataTransformer` attribute in the `PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType` class.